Framework/PythonInterface/mantid/api/src/FitFunctions/IFunctionAdapter.cpp

```diff
@@ -167,6 +167,7 @@ void IFunctionAdapter::setAttribute(const std::string &attName,
     object value = object(handle<>(getAttributeValue(*this, attr)));
     callMethod<void, std::string, object>(getSelf(), "setAttributeValue",
                                           attName, value);
+    storeAttributeValue(attName, attr);
```

Comment:
`IFunction::setAttribute` is virtual with the default implementation that just stores the new value. Subclasses may choose to override it to, for example, change the number of parameters (see `Polynomial').